### ISSUE TITLE
Question about loss computing in BartForConditionalGeneration

### ISSUE THREAD

**Issue by JamesHujy** [huggingface/transformers#6476]
I notice that in [BartForConditionalGeneration](https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_bart.py#L1043), the labels and logits are not shifted when computing cross-entropy loss. Should I provide a pre-possessed shifted labels to the model for training?

**Comment by patil-suraj** [comment]
Hi @JamesHujy , yes when training BART you need to shift `labels` and `decoder_input_ids`.

```python3
target_text = "some target text"
enc = tokenizer(target_text , return_tensors="pt")
target_ids = enc["input_ids"]
decoder_input_ids = target_ids[:, :-1].contiguous()
labels = target_ids[:, 1:].clone() 
```
